### PR TITLE
fix(aio): document XSRF protection in HttpClient

### PR DESCRIPTION
**Pull request by alxhub**
AIO only

### CODE REVIEW

**Review by IgorMinar — COMMENTED** [review]
we should also update the security doc by deleting the existing section and pointing to this http client doc. https://angular.io/guide/security#http

otherwise lgtm

File: aio/content/guide/http.md [hunk 1]
```diff
@@ -525,6 +525,44 @@ http.request(req).subscribe(event => {
 });
 ```
 
+## Security: XSRF Protection
+
+`HttpClient` comes with built-in support for [Cross-Site Request Forgery (XSRF)](https://en.wikipedia.org/wiki/Cross-site_request_forgery#Cookie-to-Header_Token) protection. Protecting against XSRF attacks requires cooperation from both the client application and the backend. Generally, the client is given a unique secret token, an XSRF token, every time it loads, and it must send the token along with every outgoing mutate request such as POST, DELETE, etc. to its backend service.
+
+`HttpClient` implements the client half of this protocol by default. It sets up an 
+interceptor which looks for a cookie set on the page named `XSRF-TOKEN`. If it finds one, 
+it send the value in the `X-XSRF-TOKEN` header on eligible outgoing requests, specifically, mutating requests to relative URLs.
```
Comment (gkalpak) [inline aio/content/guide/http.md]: _send_ --> _sends_

**Review by gkalpak — COMMENTED** [review]
For reference, here is AngularJS' description of XSRF protection: https://docs.angularjs.org/api/ng/service/$http#cross-site-request-forgery-xsrf-protection

(It might be worth adding some more info, such as that the cookie needs to be readable by JavaScript and that each app should use its own token (even for apps on the same domain/subdomain).)

File: aio/content/guide/http.md [hunk 2]
```diff
@@ -525,6 +525,48 @@ http.request(req).subscribe(event => {
...
+
+By default, an interceptor sends this cookie on all mutating requests (POST, etc.) 
+to relative URLs but not on GET/HEAD requests or 
+on requests with an absolute URL.
+
+To take advantage of this, your server needs to set a token in a JavaScript readable session cookie called `XSRF-TOKEN` on either the page load or the first GET request. On subsequent requests the server can verify that the cookie matches the `X-XSRF-TOKEN HTTP` header, and therefore be sure that only code running on your domain could have sent the request. The token must be unique for each user and must be verifiable by the server; this prevents the client from making up its own tokens. Set the token to a digest of your site's authentication
```
Comment (gkalpak) [inline aio/content/guide/http.md]: \`X-XSRF-TOKEN HTTP\` --> \`X-XSRF-TOKEN\` HTTP
(misplaced backtick)

**Review by gkalpak — APPROVED** [review]
LGTM with comments addressed (one way or another :smiley:)

**Review by juleskremer — APPROVED** [review]

**Pull request merged**